ui/src/app/registration/modal/modal.component.ts

```diff
@@ -96,30 +102,31 @@ export class RegistrationModalComponent implements OnInit {
   private getForm(role: string): FormGroup {
     if (role === 'installer') {
       return this.formBuilder.group({
-        companyName: new FormControl("", Validators.required),
-        firstname: new FormControl("", Validators.required),
-        lastname: new FormControl("", Validators.required),
-        street: new FormControl("", Validators.required),
-        zip: new FormControl("", Validators.required),
-        city: new FormControl("", Validators.required),
-        country: new FormControl("", Validators.required),
-        phone: new FormControl("", Validators.required),
-        email: new FormControl("", [Validators.required, Validators.email]),
-        password: new FormControl("", Validators.required),
-        confirmPassword: new FormControl("", Validators.required),
+        companyName: new FormControl("", [Validators.required, Validators.maxLength(50)]),
+        firstname: new FormControl("", [Validators.required, Validators.maxLength(50)]),
+        lastname: new FormControl("", [Validators.required, Validators.maxLength(50)]),
+        street: new FormControl("", [Validators.required, Validators.maxLength(50)]),
+        zip: new FormControl("", [Validators.required, Validators.pattern('^[0-9]{5}$')]),
```

Comment:
This requires 5 digits, right? That won't work e.g. for Austria. Can we find a better regex - ideally including the source webpage/discussion?